src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SmkRulesAndCheckpointsCompletionContributor.kt

```diff
@@ -26,18 +25,18 @@ class SmkRulesAndCheckpointsCompletionContributor : CompletionContributor() {
     init {
         extend(
                 CompletionType.BASIC,
-                SmkRuleNameReferenceCompletionProvider.IN_SMK_LOCALRULES_OR_RULEORDER_RULE_NAME_REFERENCE,
-                SmkRuleNameReferenceCompletionProvider()
+                SmkRuleAndCheckpointNameReferenceCompletionProvider.IN_SMK_LOCALRULES_OR_RULEORDER_RULE_NAME_REFERENCE,
+                SmkRuleAndCheckpointNameReferenceCompletionProvider()
         )
         extend(
                 CompletionType.BASIC,
-                SmkRulesAndCheckpointsObjectsCompletionProvider.IN_SMK_RULES_OR_CHECKPOINTS_OBJECT,
-                SmkRulesAndCheckpointsObjectsCompletionProvider()
+                SmkRulesOrCheckpointsObjectsCompletionProvider.IN_SMK_RULES_OR_CHECKPOINTS_OBJECT,
+                SmkRulesOrCheckpointsObjectsCompletionProvider()
         )
     }
 }
```

Comment:
Here filler name better describes the provider idea. You have 2 providers, one provides completion in `localrules` and `ruleorder` sections, the other one after `rules` and `checkpoints` objects. This information in provider name is easier to read/understand compared to `rules or checkpoints provider` and `rule and checkpoint provider` which look very similar. (no action requied, i've renamed)